lab_2_tokenize_by_bpe/main.py

```diff
@@ -14,6 +14,23 @@ def prepare_word(
     :param end_of_word: a token that signifies the end of word
     :return: preprocessed word
     """
+    if not (isinstance(raw_word, str)
+            and isinstance(start_of_word, (str, type(None)))
```

Comment:
This is a smart workaround, however it is better to check whether the variable is `None` by directly stating `<var> is None` (exactly `is`! can you tell why?)